gst/mqtt/mqttsink.c

```diff
@@ -764,6 +772,27 @@ gst_mqtt_sink_event (GstBaseSink * basesink, GstEvent * event)
   return ret;
 }
 
+/**
+ * @brief An implementation of the set_caps vmethod in GstBaseSinkClass
+ */
+static gboolean
+gst_mqtt_sink_set_caps (GstBaseSink * basesink, GstCaps * caps)
+{
+  GstMqttSink *self = GST_MQTT_SINK (basesink);
+  gboolean ret;
+
+  ret = gst_caps_replace (&self->in_caps, caps);
+
+  if (ret && gst_caps_is_fixed (self->in_caps)) {
+    char *caps_str = gst_caps_to_string (caps);
+
+    strncpy (self->mqtt_msg_hdr.gst_caps_str, caps_str, strlen (caps_str));
```

Comment:
min(strlen(caps_str), GST_MQTT_MAX_LEN_GST_CPAS_STR-1) ?

Reply:
That would be better. Thanks.